### PR TITLE
Issue #756: Import EAF files corpus frequency functionality.

### PR DESCRIPTION
**Pull request by susanodd**
Functionality to import EAF files to create new corpora.
Includes extensive testing.

### CODE REVIEW

**Comment by susanodd** [comment]
The tests make use of new folders in the test_data folder.

writable/test_data/eafs/
writable/test_data/metadata_eafs/

For example, using NGT as a test dataset for testing, recall that the test dataset is initially empty and has no glosses.
The following is a setup for using NGT as a test dataset:

writable/test_data/metadata_eafs/NGT_metadata.csv
writable/test_data/eafs/NGT/CNGT1008.eaf
etc.

The "create copus" functionality looks at the size of the EAF files.
Files larger than 50000 bytes are processed sequentially, smaller files in one batch.
Additional feedback will be added on what glosses were not found in the Dataset during processing.
Sequentially processing files allows feedback to be generated per document.

**Comment by susanodd** [comment]
This branch has undergone many more revisions. All of the frequency functions are in a file frequency.py now.
Plus there is also functionality to update a single document. Issue #756 shows screenshots.
The processing of the metadata file now has its own button in the template.

The tests are working on all the new code.

Left to determine: which sample eaf files to choose for the test_data/eafs folder?
At the moment, I've got all the CNGT files there as I've been checking how much time this consumes.
Hence the introduction of functionality to update one document.

**Comment by susanodd** [comment]
I revised the Corpus Overview Template to use Ajax calls for updating the documents shown in the list. Each row has a button (as shown in the screenshots of #756), so now those are Ajax calls. This has the advantage that the entire page is not reloaded, just the result of the call.
I modified the columns to show feedback in the last column. So that's where it shows up if no annotations were found.
It's much speeded up using Ajax.
I'm wondering about the Create Corpus and Update All Documents buttons. Those are slow.

**Review by Woseseltops — APPROVED** [review]
I rereviewed this one and I'm happy: this code is clean, organized and easily readable. The minor things I saw (that always appear when programmer A looks at code by programmer B) are not worth delaying the merge over.

As a general remark: you explain several (in my view slightly unusual) coding style decisions above. I think it would be good to summarize such argumentation in a single comment line directly above the line it is about :) .
